### DISCUSSION TITLE
Forward and Backward Animation

### DISCUSSION THREAD

**Discussion by luka5** [slidevjs/slidev#1159] [Q&A]
Hi,

I like using slidev a lot, usually it fits most of my requirements. Now I'm wondering, how I can achieve the following: On one slide, click through a custom animation, in both directions: forward and backward. In my use-case, I want to zoom into a picture and move from left to right on forward click and right to left on backward click.

I managed to utilize [element transitions](https://sli.dev/guide/animations.html#element-transitions) and [Motion](https://sli.dev/guide/animations.html#motion) - also in combination with `v-if="$slidev.nav.clicks === 1"` as mentioned in https://github.com/slidevjs/slidev/discussions/155 . This works fine, but it show the same animation regardless the direction. I'd like to use different animations depending on the direction.

I was wondering if there is some way to achieve this, maybe with one of these ideas:
* define different animations via css classes
* achieve this with `@vueuse/motion`
  * I've seen `onComplete` there, this would work - but without the click.
* implement some custom `v-click` callback function, where I can define and run the animation depending on the previous and current "click".

Thanks a lot

**Comment by kermanx** [comment]
Now we have [Direction Specific Animations](https://sli.dev/guide/animations#direction-specific-animations) and [Motion with Clicks](https://sli.dev/guide/animations#motion-with-clicks)

**Reply by c-harding** [reply]
Updated link: [Direction Specific Animations](https://sli.dev/features/direction-variant)
